mllib/src/main/scala/org/apache/spark/mllib/regression/StreamingLinearRegressionWithSGD.scala

```diff
@@ -59,6 +59,8 @@ class StreamingLinearRegressionWithSGD private[mllib] (
 
   val algorithm = new LinearRegressionWithSGD(stepSize, numIterations, miniBatchFraction)
 
+  var model: Option[LinearRegressionModel] = None
```

Comment:
Thanks for reviewing @srowen , I'm a little puzzled myself, but undoing either one of these changes definitely causes failures in the newly added tests. My guess is it has something to do with how the closure cleaner handles these vars, which may be affected both by the declaration and the function def, but will need to dig in more to understand fully. Hoping @tdas might have some ideas as well.